planner/optimizer/optimizer-core/src/test/java/eu/seaclouds/platform/planner/optimizerTest/OptimizerTOSCADecember2015AutoscalingPoliciesTest.java

```diff
@@ -0,0 +1,240 @@
+/**
+ * Copyright 2015 SeaClouds
+ * Contact: SeaClouds
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package eu.seaclouds.platform.planner.optimizerTest;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Map;
+
+import org.junit.Assert;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.yaml.snakeyaml.Yaml;
+
+import eu.seaclouds.platform.planner.optimizer.Optimizer;
+import eu.seaclouds.platform.planner.optimizer.heuristics.SearchMethodName;
+import eu.seaclouds.platform.planner.optimizer.util.TOSCAkeywords;
+import eu.seaclouds.platform.planner.optimizer.util.YAMLgroupsOptimizerParser;
+import eu.seaclouds.platform.planner.optimizer.util.YAMLmodulesOptimizerParser;
+import eu.seaclouds.platform.planner.optimizer.util.YAMLoptimizerParser;
+import eu.seaclouds.platform.planner.optimizer.util.YAMLtypesOptimizerParser;
+
+public class OptimizerTOSCADecember2015AutoscalingPoliciesTest {
+
+   private static Optimizer optimizer;
+   private static String    appModel;
+   private static String    suitableCloudOffer;
+
+   static Logger log;
+
+   @BeforeClass
+   public void createObjects() {
+
+      log = LoggerFactory.getLogger(OptimizerTOSCADecember2015AutoscalingPoliciesTest.class);
+
+      log.info("Starting TEST optimizer for the TOSCA syntax of September 2015");
+
+      final String dir = System.getProperty("user.dir");
+      log.debug("Trying to open files: current executino dir = " + dir);
+
+      try {
+         appModel = filenameToString(TestConstants.APP_MODEL_FILENAME);
+      } catch (IOException e) {
+         log.error("File for APPmodel not found");
+         e.printStackTrace();
+      }
+
+      try {
+         suitableCloudOffer = filenameToString(TestConstants.CLOUD_OFFER_FILENAME_IN_JSON);
+      } catch (IOException e) {
+         log.error("File for Cloud Offers not found");
+         e.printStackTrace();
+      }
+
+   }
+
```

Comment:
I think this code is repeated by `OptimizerComputeTypeTest`. You could create a super class for avoiding duplicate code.